poseidon2/roundfun.nim

```diff
@@ -0,0 +1,48 @@
+import
+  constantine/math/arithmetic,
+  constantine/math/config/curves
+
+import ./types
+import ./roundconst
+
+#-------------------------------------------------------------------------------
+
+const externalRoundConst : array[24, F] = arrayFromHex( externalRoundConstStr )
+const internalRoundConst : array[56, F] = arrayFromHex( internalRoundConstStr )
+
+#-------------------------------------------------------------------------------
+
+# inplace sbox, x => x^5
+func sbox*(x: var F) : void =
```

Comment:
This could be `templates` unless. Same for the rest of the `funcs` in this file.

Reply:
I did some earlier profiling of with this and found that changing some of these to templates could actually slow things down. Using `func` lets the C compiler decide when to inline.

Reply:
You're right, the compiler might be smarter about this.

Reply:
Yah seems the machines won't leave us any areas where they're not smarter. ;)